### DISCUSSION TITLE
pocketbase and sveltkit help

### DISCUSSION THREAD

**Discussion by Tyjust31** [pocketbase/pocketbase#5859] [Q&A]
hi everyone I'm doing a project and I can't connect the user in pocketbase, here's the code                       
_hook/server/ts_
import type { Handle } from '@sveltejs/kit';
import { locale } from 'svelte-i18n';
import PocketBase from 'pocketbase';
import { PUBLIC_POCKETBASE_URL } from '$env/static/public';
import { dev } from '$app/environment';

export const handle: Handle = async ({ event, resolve }) => {
  // Set locale from Accept-Language header
  const lang = event.request.headers.get('accept-language')?.split(',')[0];
  if (lang) {
    locale.set(lang);
  }
 

  event.locals.pb = new PocketBase('http://127.0.0.1:8090');

    // load the store data from the request cookie string
    event.locals.pb.authStore.loadFromCookie(event.request.headers.get('cookie') || '');

    try {
        // get an up-to-date auth store state by verifying and refreshing the loaded auth model (if any)
        event.locals.pb.authStore.isValid && await event.locals.pb.collection('users').authRefresh();
    } catch (_) {
        // clear the auth store on failed refresh
        event.locals.pb.authStore.clear();
    }

    const response = await resolve(event);

    // send back the default 'pb_auth' cookie to the client with the latest store state
    response.headers.append('set-cookie', event.locals.pb.authStore.exportToCookie());

    return response;

};
_login/page.server.ts_
import { error } from '@sveltejs/kit';
import PocketBase from 'pocketbase';
import { fail, redirect } from '@sveltejs/kit';
import type { Actions } from '@sveltejs/kit';
import type { ClientResponseError } from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

export const actions: Actions = {
    login: async ({ request, cookies }: { request: Request, cookies: any }) => {
        const formData = await request.formData();
        const email = formData.get('email') as string;
        const password = formData.get('password') as string;

        // Validation des champs
     /*   if (!email || email.trim() === '') {
            return fail(400, { email, missing: true, message: 'Email is required' });
        }

        const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
        if (!emailPattern.test(email)) {
            return fail(400, { email, incorrect: true, message: 'Invalid email address' });
        }

        if (!password || password.trim() === '') {
            return fail(400, { password, missing: true, message: 'Password is required' });
        }

        if (password.length < 8 || password.length > 32) {
            return fail(400, {
                password,
                incorrect: true,
                message: 'Password must be between 8 and 32 characters'
            });
        }*/

        // Tentative de connexion
        try {
         const authData = await pb.collection('users').authWithPassword(email, password);
         if (!pb?.authStore?.model?.verified) {
            pb.authStore.clear();
            console.log('Votre compte n\'est pas vérifié');
            return {
                notVerified: true,
                message: 'Votre compte n\'est pas vérifié',
               
            };
        }

        console.log( 'token' , pb?.authStore?.isValid.valueOf());
            
            // Si la connexion réussit, stocker le token d'authentification dans un cookie
        /*    cookies.set('pb_auth', pb.authStore.token, {
                path: '/',
                secure: process.env.NODE_ENV === 'production',
                httpOnly: true,
                sameSite: 'strict',
                maxAge: 60 * 60 * 24 * 30 // 30 jours
            });*/

            // Redirection si l'authentification réussit
            console.log('Authentification réussie');
            console.log(authData);
            throw redirect(307, '/organization');
        } catch (err) {
                console.error('Erreur de connexion:', err);
                return fail(400, {
                    email,
                    password,
                    incorrect: true,
                    message:  'Invalid email or password'
                });
        
        }
    }};

**Comment by Tyjust31** [comment]
Here's the result I get:

// token true
Authentification réussie
{
  record: {
    avatar: "",
    collectionId: "_pb_users_auth_",
    collectionName: "users",
    created: "2024-11-13 08:50:18.671Z",
    email: "gg@gg.com",
    emailVisibility: false,
    first_name: "",
    id: "e7akkc53nfqzsbh",
    job: "",
    last_name: "",
    updated: "2024-11-13 19:49:09.050Z",
    username: "ggg",
    verified: true,
  },
  token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF8iLCJleHAiOjE3MzI3NzUzODAsImlkIjoiZTdha2tjNTNuZnF6c2JoIiwidHlwZSI6ImF1dGhSZWNvcmQifQ.0bnk6QeEsCeC9v0AYkxHEQhO8heMNz9Dz8mQnpAuSMg",
}
Erreur de connexion: Redirect {
  status: 307,
  location: "/organization",
}

**Comment by ganigeorgiev** [comment]
Please don't open questions like this.

I consider it lazy and low-effort to pile bunch of unformatted code and have just one sentence that doesn't really explain what is the problem or the desired behavior.

From a brief look, if the problem is that your code ended up calling the `catch` block then this is expected because you have `throw redirect(307, '/organization')`.

In any case, please don't treat the discussions and issues tracker as AI chat and try to:

- elaborate what actually is the problem
- what is the desired behavior
- try to isolate your code with only the minimal necessary bits and format it with 3 backticks ([Creating and highlighting code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))

**Reply by Tyjust31** [reply]
I've been using pocketbase and sveltkit for several days but the connection doesn't work. I've tried to solve the problem and I've done everything I can to register, but I can't get the connection to work. I have tried CORS and nothing with curl yes I get validation but in my code base nothing . The data is sent but the authentication doesn't work and people with the same problem never get a response.

**Reply by Tyjust31** [reply]
I know we can be frustrated by certain messages, but I have an error. I know how git hub works, but on this account I can't concatenate the code.  if you dont have a solution for me it's nothing

**Comment by Tyjust31** [comment]
`            import { error } from '@sveltejs/kit';
import PocketBase from 'pocketbase';
import { fail, redirect } from '@sveltejs/kit';
import type { Actions } from '@sveltejs/kit';
import type { ClientResponseError } from 'pocketbase';



const pb = new PocketBase('http://127.0.0.1:8090');

export const actions: Actions = {
    login: async ({ request, cookies }: { request: Request, cookies: any }) => {
        const formData = await request.formData();
        const email = formData.get('email') as string;
        const password = formData.get('password') as string;

        // Validation des champs
     /*   if (!email || email.trim() === '') {
            return fail(400, { email, missing: true, message: 'Email is required' });
        }

        const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
        if (!emailPattern.test(email)) {
            return fail(400, { email, incorrect: true, message: 'Invalid email address' });
        }

        if (!password || password.trim() === '') {
            return fail(400, { password, missing: true, message: 'Password is required' });
        }

        if (password.length < 8 || password.length > 32) {
            return fail(400, {
                password,
                incorrect: true,
                message: 'Password must be between 8 and 32 characters'
            });
        }*/

        // Tentative de connexion
        try {
         const authData = await pb.collection('users').authWithPassword(email, password);
         if (!pb?.authStore?.model?.verified) {
            pb.authStore.clear();
            console.log('Votre compte n\'est pas vérifié');
            return {
                notVerified: true,
                message: 'Votre compte n\'est pas vérifié',
               
            };
        }

        console.log( 'token' , pb?.authStore?.isValid.valueOf());
            
            // Si la connexion réussit, stocker le token d'authentification dans un cookie
        /*    cookies.set('pb_auth', pb.authStore.token, {
                path: '/',
                secure: process.env.NODE_ENV === 'production',
                httpOnly: true,
                sameSite: 'strict',
                maxAge: 60 * 60 * 24 * 30 // 30 jours
            });*/

            // Redirection si l'authentification réussit
            console.log('Authentification réussie');
            console.log(authData);

           if (authData){
            throw redirect(307, '/organization');
           }
       
        } catch (err) {
                console.error('Erreur de connexion:', err);
                return fail(400, {
                    email,
                    password,
                    incorrect: true,
                    message:  'Invalid email or password'
                });
        
        }
    }};                        `
